torch/_dynamo/variables/optimizer.py

```diff
@@ -310,13 +310,15 @@ def mark_static(x):
             p_state_source = GetItemSource(
                 state_source, ConstDictKeySource(state_source, idx)
             )
-            for k, v in value.items():
+            for inner_idx, (k, v) in enumerate(value.items()):
                 if (
                     isinstance(v, torch.Tensor)
                     and v not in self.grad_to_source
                     and v not in self.tensor_to_source
                 ):
-                    self.tensor_to_source[v] = GetItemSource(p_state_source, k)
+                    self.tensor_to_source[v] = GetItemSource(
```

Comment:
You might just need something like this here - https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/variables/dicts.py#L297

Reply:
Since this is not going through VariableBuilder, we have to manually mark things here - https://github.com/pytorch/pytorch/blob/dcf22fa58cb63c3762c67e66ef85a9e564703ebb/torch/_dynamo/variables/builder.py#L616-L637

Reply:
Yeah it looks like it is going through VB, so there must be something else going on.